gunrock/app/geo/geo_d_spatial.cuh

```diff
@@ -0,0 +1,354 @@
+// ----------------------------------------------------------------
```

Comment:
(low priority) Is it possible to combine this file w/ `geo_spatial.cuh`?  (That's an actual question, not a rhetorical question).  Would reduce the amount of code significantly, esp. since these spatial helper functions aren't super obvious

Reply:
Also, I didn't check every line of the math here, since you said you're confident in the correctness testing

Reply:
Yes, I think I can combine the two files if I use `Array1D` representation for host arrays as well. I'll try to merge it together to reduce code.

Reply:
Done!